nat-lab/tests/conftest.py

```diff
@@ -197,24 +197,28 @@ def pytest_make_parametrize_id(config, val):
 
 
 async def setup_check_interderp():
-    async with new_connection_raw(ConnectionTag.DOCKER_CONE_CLIENT_1) as connection:
-        if not isinstance(connection, DockerConnection):
-            raise Exception("Not docker connection")
-        containers = [
-            connection.container_name(),
-            "nat-lab-derp-01-1",
-            "nat-lab-derp-02-1",
-            "nat-lab-derp-03-1",
+    async with AsyncExitStack() as exit_stack:
+        connections = [
+            await exit_stack.enter_async_context(new_connection_raw(conn_tag))
+            for conn_tag in [
+                ConnectionTag.DOCKER_CONE_CLIENT_1,
+                ConnectionTag.DOCKER_DERP_1,
+                ConnectionTag.DOCKER_DERP_2,
+                ConnectionTag.DOCKER_DERP_3,
+            ]
         ]
-        start_tcpdump(containers)
-        try:
+
+        if not isinstance(connections[0], DockerConnection):
+            raise Exception("Not docker connection")
+
+        async with make_tcpdump(connections):
```

Comment:
I am thinking, maybe it would make sense to combine packet capture started in `setup_check_interderp()` with the general one started in `pytest_sessionstart`? Or are those non-overlapping in time? E.g. by adding DERP servers to the list here: https://github.com/NordSecurity/libtelio/pull/873/files#diff-0433c1e25b1d0e1b02e84b811ebf3db74a6d26d59f6439e210b242377d92836aR421

Reply:
Note this is rather a nitpick, so it is up to you to decide whether this is relevant enough to fix.